tests/functional/olp-cpp-sdk-dataservice-read/ApiTest.cpp

```diff
@@ -0,0 +1,455 @@
+#include "generated/api/PlatformApi.h"
+#include "generated/api/QueryApi.h"
+#include "generated/api/VolatileBlobApi.h"
+
+class ApiTest : public ::testing::Test {
+ protected:
+  ApiTest() {
+    auto network = olp::client::OlpClientSettingsFactory::
+        CreateDefaultNetworkRequestHandler();
+
+    olp::authentication::Settings authentication_settings;
+    authentication_settings.network_request_handler = network;
+
+    olp::authentication::TokenProviderDefault provider(
+        CustomParameters::getArgument("dataservice_read_test_appid"),
+        CustomParameters::getArgument("dataservice_read_test_secret"),
+        authentication_settings);
+    olp::client::AuthenticationSettings auth_client_settings;
+    auth_client_settings.provider = provider;
+
+    settings_ = std::make_shared<olp::client::OlpClientSettings>();
+    settings_->authentication_settings = auth_client_settings;
+    settings_->network_request_handler = network;
+
+    client_ = olp::client::OlpClientFactory::Create(*settings_);
+  }
+  std::string GetTestCatalog() {
+    return CustomParameters::getArgument("dataservice_read_test_catalog");
+  }
+
+  std::string PrintError(const olp::client::ApiError& error) {
+    std::ostringstream resultStream;
+    resultStream << "ERROR: status: " << error.GetHttpStatusCode()
+                 << ", message: " << error.GetMessage();
+    return resultStream.str();
+  }
+
+ protected:
+  std::shared_ptr<olp::client::OlpClientSettings> settings_;
+  std::shared_ptr<olp::client::OlpClient> client_;
+};
+
+TEST_F(ApiTest, GetCatalog) {
+  olp::client::HRN hrn(GetTestCatalog());
+
+  std::promise<olp::dataservice::read::ApiClientLookup::ApiClientResponse>
+      configClientPromise;
+  olp::dataservice::read::ApiClientLookup::LookupApiClient(
+      olp::client::OlpClientFactory::Create(*settings_), "config", "v1", hrn,
+      [&configClientPromise](
+          olp::dataservice::read::ApiClientLookup::ApiClientResponse response) {
+        configClientPromise.set_value(response);
+      });
+
+  auto clientResponse = configClientPromise.get_future().get();
+  ASSERT_TRUE(clientResponse.IsSuccessful())
+      << PrintError(clientResponse.GetError());
+  auto configClient = clientResponse.GetResult();
+
+  std::promise<olp::dataservice::read::ConfigApi::CatalogResponse>
+      catalogPromise;
+  auto catalogCallback =
+      [&catalogPromise](
+          olp::dataservice::read::ConfigApi::CatalogResponse catalogResponse) {
+        catalogPromise.set_value(catalogResponse);
+      };
+
+  auto startTime = std::chrono::high_resolution_clock::now();
+  auto cancelFn = olp::dataservice::read::ConfigApi::GetCatalog(
+      configClient, GetTestCatalog(), boost::none, catalogCallback);
+  auto catalogResponse = catalogPromise.get_future().get();
+  auto end = std::chrono::high_resolution_clock::now();
+
+  std::chrono::duration<double> time = end - startTime;
+  std::cout << "duration: " << time.count() * 1000000 << " us" << std::endl;
```

Comment:
I would replace std::cout with trace

Reply:
I recall we decided to not use sdk logs in tests

Reply:
For some platforms the `std::cout` won't produce anything